src/Equinox/Decider.fs

```diff
@@ -108,6 +108,39 @@ type Decider<'event, 'state>(inner : DeciderCore<'event, 'state>) =
         let inline decide' c ct = task { let! r, es = Async.StartImmediateAsTask(decide c, ct) in return struct (r, Seq.ofList es) }
         return! inner.TransactExAsync(decide = decide', mapResult = mapResult, ?load = load, ?attempts = attempts, ct = ct) |> Async.AwaitTaskCorrect }
 
+    /// 1. Invoke the supplied <c>Task</c> returning <c>interpret</c> function with the present state
+    /// 2. (if events yielded) Attempt to sync the yielded events to the stream.
+    ///    (Restarts up to <c>maxAttempts</c> times with updated state per attempt, throwing <c>MaxResyncsExhaustedException</c> on failure of final attempt.)
+    /// 3. Uses <c>render</c> to generate a 'view from the persisted final state
+    member _.TransactTask(interpret : 'state -> CancellationToken -> Task<'event list>, render : 'state -> 'view, ?load, ?attempts, ?ct) : Task<'view> =  task {
+        let inline interpret' s ct = task { let! es = interpret s ct in return Seq.ofList es }
+        return! inner.TransactAsync(interpret', render, ?load = load, ?attempts = attempts, ?ct = ct) }
+
+    /// 1. Invoke the supplied <c>Task</c> returning <c>decide</c> function with the present state, holding the <c>'result</c>
+    /// 2. (if events yielded) Attempt to sync the yielded events to the stream.
+    ///    (Restarts up to <c>maxAttempts</c> times with updated state per attempt, throwing <c>MaxResyncsExhaustedException</c> on failure of final attempt.)
+    /// 3. Yield result
+    member _.TransactTask(decide : 'state -> CancellationToken -> Task<'result * 'event list>, ?load, ?attempts, ?ct) : Task<'result> =  task {
```

Comment:
what would you think about forcing `decide` to return struct tuples? and maybe `seq` ?

Reply:
> `decide` to return struct tuples?

I'd love it if F# removed the need to add the `struct` keyword from the calling side (does it do that yet?). I've noticed the struct tuple / tuple divide cause confusion in my team at least. Though yes, I do agree that a struct tuple is appropriate here. I'm torn 🤔 

> and maybe `seq`?

I think that'd go against the programming model. We encourage users to set up deciders in a certain way that uses lists, and as such I don't think moving the `seq` transformation to the client provides a lot of value.